stock_picking_order_alert/models/stock_picking.py

```diff
@@ -0,0 +1,152 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    has_quantity_alert = fields.Boolean(
+        string="Quantity Alert",
+        compute="_compute_has_quantity_alert",
+        store=False,
+        help="Indicates if any move line exceeds ordered "
+        "quantity by more than threshold",
+    )
+    quantity_alert_message = fields.Html(
+        string="Quantity Alert Message",
+        compute="_compute_has_quantity_alert",
+        store=False,
+    )
+
+    def _is_quantity_alert_enabled(self):
+        """Check if the quantity alert feature is enabled based on settings:
+        1. First checks if display_quantity_alert_percentage is enabled (True)
+        2. Then checks if alert_percentage threshold is greater than zero
+        """
+        IrConfigParam = self.env["ir.config_parameter"].sudo()
+
+        # First check if the alert feature is enabled in the settings
+        display_enabled = bool(
+            IrConfigParam.get_param(
+                "stock_picking_order_alert.display_quantity_alert_percentage",
+                default=False,
+            )
+        )
+
+        # If display is not enabled, return immediately
+        if not display_enabled:
+            return False, 0.0
+
+        # Now check the alert percentage threshold
+        alert_percentage = float(
+            IrConfigParam.get_param(
+                "stock_picking_order_alert.alert_percentage", default=30.0
+            )
+        )
+
+        # Feature is enabled if display_enabled is True and alert_percentage > 0
+        is_enabled = (
+            display_enabled
+            and float_compare(alert_percentage, 0.0, precision_digits=2) > 0
```

Comment:
```suggestion
            and alert_percentage
```

Given last modifications this should be enough